lib/internal/warning.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+// A process warning is esssentially just an Error object emitted through the
+// process.on('warning') event rather than being thrown.
+
+function Warning(name, message, constructor) {
+  const warning = new Error(message);
+  warning.name = name || 'Warning';
+  Error.captureStackTrace(warning, constructor || Warning);
+  return warning;
+}
+
+Warning.badPracticeWarning = function(message, warned) {
+  if (warned) return warned;
```

Comment:
not consistent with other styles used: https://github.com/jasnell/node/blob/cde7688d9d8daf499d5caeedc267df07c806904d/lib/internal/util.js#L21-L22